Simplified/AxisService/AxisService.swift

```diff
@@ -20,22 +20,227 @@ protocol NYPLBookDownloadBroadcasting {
+      
+    } catch {
+      print("error initiating AxisService: \(error)")
+      return nil
+    }
   }
   
   /// Fulfill AxisNow license
   /// - Parameters:
-  ///   - fileURL: Downloaded LCP license URL
   ///   - book: `NYPLBook` Book
   ///   - downloadTask: downloadTask download task
-  @objc func fulfillAxisLicense(withFileURL fileURL: URL,
-                                forBook book: NYPLBook,
+  @objc func fulfillAxisLicense(forBook book: NYPLBook,
                                 downloadTask: URLSessionDownloadTask) {
     
+    let dispatchGroup = DispatchGroup()
+    downloadLicense(group: dispatchGroup, forBook: book)
+    dispatchGroup.wait()
+    downloadEncryption(group: dispatchGroup, forBook: book)
+    dispatchGroup.wait()
+    downloadContainer(group: dispatchGroup, forBook: book)
+    dispatchGroup.wait()
+    downloadPackage(group: dispatchGroup, forBook: book)
+    dispatchGroup.wait()
+    downloadAssetsFromPackage(group: dispatchGroup, forBook: book)
+    
+    dispatchGroup.notify(queue: DispatchQueue.main) {
+      print("Book content downloaded!")
+      //      _ = self.delegate?.replaceBook(
+      //        book,
+      //        withFileAtURL: self.dedicatedWriteURL,
+      //        forDownloadTask: downloadTask)
+    }
+  }
+  
+  private func downloadLicense(group: DispatchGroup, forBook book: NYPLBook) {
+    group.enter()
+    let writeURL = dedicatedWriteURL.appendingPathComponent("license.json")
+    guard let licenseURL = AxisLicenseURLGenerator(
+      isbn: self.isbn,
+      bookVaultId: self.bookVaultId
+    ).licenseURL else {
+      return
+    }
+    
+    AxisContentDownloader().downloadContent(from: licenseURL) { (result) in
+      switch result {
+      case .success(let data):
+        do {
+          try AxisAssetWriter().writeAsset(data, atURL: writeURL)
+          group.leave()
+        } catch {
+          print(error)
+          group.leave()
+          self.delegate?.failDownloadWithAlert(forBook: book)
+        }
+      case .failure(let error):
+        print(error)
+        group.leave()
+        self.delegate?.failDownloadWithAlert(forBook: book)
+      }
+      
+    }
+  }
+  
+  private func downloadEncryption(group: DispatchGroup, forBook book: NYPLBook) {
+    group.enter()
+    let encryptionURL = baseURL.appendingPathComponent("META-INF/encryption.xml")
+    let writeURL = self.dedicatedWriteURL.appendingPathComponent("encryption.xml")
+    AxisContentDownloader().downloadContent(from: encryptionURL) { (result) in
+      switch result {
+      case .success(let data):
+        do {
+          try AxisAssetWriter().writeAsset(data, atURL: writeURL)
+          group.leave()
+        } catch {
+          print(error)
+          group.leave()
+          self.delegate?.failDownloadWithAlert(forBook: book)
+        }
+      case .failure(let error):
+        print(error)
+        group.leave()
+        self.delegate?.failDownloadWithAlert(forBook: book)
+      }
+    }
```

Comment:
This code is used multiple times in the same file. Definitely needs fixing 
`AxisContentDownloader().downloadContent(from: encryptionURL) { (result) in
      switch result {
      case .success(let data):
        do {
          try AxisAssetWriter().writeAsset(data, atURL: writeURL)
          group.leave()
        } catch {
          print(error)
          group.leave()
          self.delegate?.failDownloadWithAlert(forBook: book)
        }
      case .failure(let error):
        print(error)
        group.leave()
        self.delegate?.failDownloadWithAlert(forBook: book)
      }
    }`